File: src/first_fit.js
import Bin from './bin.js';

function first_fit_decreasing(items, bin_capacity) {
    // Sort items in decreasing order
    const sorted_items = [...items].sort((a, b) => b - a);
    
    const bins = [];
    
    for (const item of sorted_items) {
        let placed = false;
        for (const bin of bins) {
            if (bin.can_fit(item)) {
                bin.add_item(item);
                placed = true;
                break;
            }
        }
        if (!placed) {
            const new_bin = new Bin(bin_capacity);
            new_bin.add_item(item);
            bins.push(new_bin);
        }
    }
    
    return bins;
}

export { first_fit_decreasing };

File: src/first_fit.py
from bin import Bin

def first_fit_decreasing(items, bin_capacity):
    # Sort items in decreasing order
    sorted_items = sorted(items, reverse=True)
    
    bins = []
    
    for item in sorted_items:
        placed = False
        for b in bins:
            if b.can_fit(item):
                b.add_item(item)
                placed = True
                break
        if not placed:
            new_bin = Bin(bin_capacity)
            new_bin.add_item(item)
            bins.append(new_bin)
    
    return bins

